src/content/en/showcase/2016/service-worker-perf.md

```diff
@@ -0,0 +1,679 @@
+
+For IOWA, we wanted to track two additional things:
+
+* The amount of time that elapses between when the page first starts to load and when pixels appear on the screen.
+* Whether or not a service worker is controlling the page. With this information we could then segment our reports to compare the results with and without service worker.
+
+### Capturing time to first paint
+
+Some browsers record the precise time at which the first pixel is painted to the screen, and they make that time available to developers. That value, compared with the `navigationStart` value exposed via the [Navigation Timing API](https://developer.mozilla.org/en-US/docs/Web/API/Navigation_timing_API) gives us a very accurate accounting of how much time has passed between when the user initially requested the page and when they first saw something.
+
+As I already mentioned, time to first paint is an important metric to measure because it's the first point at which a user experiences the load speed of your site. It's the first impression users get, and a good first impression can positively affect the rest of the user experience.&nbsp;<sup><a href="#footnotes">[2]</a></sup>
+
+To get the first paint value in browsers that expose it, we created the `getTimeToFirstPaintIfSupported` utility function:
+
+<pre class="prettyprint">{% htmlescape %}
+function getTimeToFirstPaintIfSupported() {
+  // Ignores browsers that don't support the Performance Timing API.
+  if (window.performance && window.performance.timing) {
+    var navTiming = window.performance.timing;
+    var navStart = navTiming.navigationStart;
+    var fpTime;
+
+    // If chrome, get first paint time from `chrome.loadTimes`.
+    if (window.chrome && window.chrome.loadTimes) {
+      fpTime = window.chrome.loadTimes().firstPaintTime * 1000;
+    }
+    // If IE/Edge, use the prefixed `msFirstPaint` property.
+    // See http://msdn.microsoft.com/ff974719
+    else if (navTiming.msFirstPaint) {
+      fpTime = navTiming.msFirstPaint;
+    }
+
+    if (fpTime && navStart) {
+      return fpTime - navStart;
+    }
+  }
+}
+{% endhtmlescape %}</pre>
+
+<aside class="caution">
+  <strong>Important:</strong> the <code>firstpaint</code> values referenced in the above function are part of vendor-specific APIs. They are non-standard and subject to change (or removal) at any time. We chose to implement them in IOWA knowing the site would have a limited lifespan. For most sites, it's best to use other <a href="https://speedcurve.com/blog/user-timing-and-custom-metrics/">User Timing techniques</a> until a <code>firstpaint</code> (or a <code>firstpaint</code> alternative) is standardized.
```

Comment:
What I'm referencing here is an official web API, so you still think it shouldn't be capitalized?
https://developer.mozilla.org/en-US/docs/Web/API/User_Timing_API